server/application/application.go

```diff
@@ -1826,7 +1826,20 @@ func (s *Server) ListLinks(ctx context.Context, req *application.ListAppLinksReq
 		return nil, fmt.Errorf("failed to read application deep links from configmap: %w", err)
 	}
 
-	finalList, errorList := deeplinks.EvaluateDeepLinksResponse(*obj, deepLinks)
+	clst, err := s.db.GetCluster(ctx, a.Spec.Destination.Server)
+	if err != nil {
```

Comment:
App might be configured with non-existing cluster. I think if error is `codes.NotFound` then API still need to populate links but using empty map as a cluster.

Reply:
But the link generation will still fail if the user tries to access cluster value in the template